code/app/src/main/java/com/example/bread/repository/MoodEventRepository.java

```diff
@@ -74,160 +70,80 @@ public void fetchEventsWithParticipantRef(@NonNull DocumentReference participant
+
+            // Handle the case of no followed users
             if (following == null || following.isEmpty()) {
-                // If not following anyone, return empty list
-                onSuccessListener.onSuccess(new ArrayList<>());
+                onSuccessListener.onSuccess(allMoodEvents);
                 return;
             }
 
-            // Create a list to hold all participant references
-            List<DocumentReference> participantRefs = new ArrayList<>();
-            for (String followingUsername : following) {
-                participantRefs.add(participantRepository.getParticipantRef(followingUsername));
-            }
+            // Counter to track when all queries are complete
+            final int[] queriesRemaining = {following.size()};
 
-            // If following list is large, limit it to prevent performance issues
-            if (participantRefs.size() > maxFollowing) {
-                participantRefs = participantRefs.subList(0, maxFollowing);
+            for (String followingUsername : following) {
+                // Don't use orderBy to avoid Firestore index requirements
+                getMoodEventCollRef()
+                        .whereEqualTo("participantRef", participantRepository.getParticipantRef(followingUsername))
+                        .limit(MAX_EVENTS_PER_USER)
+                        .get()
+                        .addOnSuccessListener(queryDocumentSnapshots -> {
+                            List<MoodEvent> moodEvents = new ArrayList<>();
+
+                            // Process each document, setting the ID explicitly
+                            for (DocumentSnapshot doc : queryDocumentSnapshots) {
+                                MoodEvent event = doc.toObject(MoodEvent.class);
+                                if (event != null) {
+                                    event.setId(doc.getId());
+                                    moodEvents.add(event);
+                                }
+                            }
+
+                            // Add to our combined results
+                            synchronized (allMoodEvents) {
+                                allMoodEvents.addAll(moodEvents);
+                            }
+
+                            // If all queries are complete, return the results
+                            if (queriesRemaining[0]-- <= 1) {
+                                onSuccessListener.onSuccess(allMoodEvents);
+                            }
+                        })
+                        .addOnFailureListener(e -> {
+                            Log.e(TAG, "Error fetching events for " + followingUsername, e);
+
+                            // Even on failure, decrement counter and check if done
+                            if (queriesRemaining[0]-- <= 1) {
+                                onSuccessListener.onSuccess(allMoodEvents);
+                            }
+                        });
```

Comment:
Sorry why again are changing this logic, this basically breaks the 3 latest events logic? We need indexing, if you are testing locally please enable it in your firebase